### PR TITLE
Implement ExactNNS as an agg function that can be called in trino

### PR DESCRIPTION
**Pull request by TiannanSha**
This PR first implements the support on the trino connector side for integrating the pixels `vector` type as the underlying data source for trino `array(double)`.

This PR then implements distributed Exact Nearest Neighbour search as an aggregation function `exact_nns_agg(arr_col, inputVec, distFunc, k)` that can be invoked in trino, utilising multi-node cluster. `arr_col` is the array column, input vec is the vector for which we want to find the nearest neighbors, distFunc can be euclidean distance, dot product and cosine similarity, and `k` is to specify that we want the top k neighbors.  

This PR also implements three different types of distance calculation as trino udfs: euclidean distance, dot product and cosine similarity that can be called on comparing distance on two vectors;

### CODE REVIEW

**Comment by TiannanSha** [comment]
maybe make more sense to merge this until I finish exacNNS implementation

**Comment by TiannanSha** [comment]
Ready for review now

File: connector/src/main/java/io/pixelsdb/pixels/trino/vector/VectorUDF.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        if (trinoVector == null) {
+            return null;
+        }
+        inputVector = new double[trinoVector.getPositionCount()];
+        for (int i = 0; i < trinoVector.getPositionCount(); i++) {
+            inputVector[i] = DOUBLE.getDouble(trinoVector, i);
+        }
+
+        // prepare list of files to read
+        // todo how to go from (tableId, columnId) to the list of files belong to that column?
```
Comment (TiannanSha) [inline connector/src/main/java/io/pixelsdb/pixels/trino/vector/VectorUDF.java:39]: @bianhq Maybe you have some insights on this? 🤔

**Review by bianhq — APPROVED** [review]
LGTM.

**Pull request merged**